swap/src/protocol/alice/event_loop.rs

```diff
@@ -307,11 +307,14 @@ impl LatestRate for FixedRate {
     }
 }
 
-impl LatestRate for kraken::RateUpdateStream {
+impl LatestRate for kraken::PriceUpdates {
```

Comment:
❓If I understand it correctly you renamed everything from Kraken to `price`, but once you apply the spread it is a `rate`. Additionally, a `SpreadRate` was introduced. Is there any reasoning for the distinction between price, rate and spread-rate?

Reply:
Yes.

So the idea was:

1. Kraken gives us price updates.
2. A `Rate` is what we use to calculate our quotes. With the addition of the spread, the rate is a product of a (market) price plus the spread.
3. The `SpreadRate` is a sister-type of the `FixedRate`. Both have an implementation of `LatestRate` that allows to compute whatever the current latest rate is.